### PR TITLE
Improvement on the redirection layer

### PR DESCRIPTION
**Pull request by cyrilf**
## Improvements

- **Redirect method** has been improved. It sets the `Location` header into the bag rather than directly applying an `header('Location:..');`. This allows us to go back up into all the layers.

_thanks @oliviermadre for the help_

### CODE REVIEW

File: src/Layer/RedirectionLayer.php [hunk 1]
```diff
@@ -29,11 +31,11 @@ public function setUrlGenerator(UrlGeneratorInterface $urlGenerator)
     public function handle(ResponseBag $responseBag)
     {
         if (count($this->config) == 1 && array_key_exists(0, $this->config)) {
-            if ($this->config[0][0] == '@') {
+            if ($this->config[0][0] === self::MAGIC_KEY_ROUTE_REFERENCE) {
                 $url = $this->urlGenerator->generate(substr($this->config[0], 1));
-                $this->redirect($url);
+                $this->redirect($url, $bag);
```
Comment (Keirua) [inline src/Layer/RedirectionLayer.php]: $responseBag non ? Ainsi que partout ailleurs ?

Comment (cyrilf) [inline src/Layer/RedirectionLayer.php]: Done. Thanks!

**Comment by vianneyb** [comment]
Could be the occasion to write some doc on pyrite, starting by the RedirectionLayer...

**Comment by cyrilf** [comment]
You're right, I'll create a draft

**Comment by cyrilf** [comment]
@vianneyb here it is: [Redirection layer](https://github.com/Evaneos/pyrite/wiki/Layer:-Redirection-Layer) (still in a draft version)

**Pull request merged**